spock/src/main/java/org/jboss/weld/spock/EnableWeld.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2018, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.weld.spock;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.jboss.weld.spock.impl.EnableWeldExtension;
+import spock.lang.Shared;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static org.jboss.weld.spock.EnableWeld.Scope.ITERATION;
+
+/**
+ * An annotation with which Weld can be enabled on specification or feature level if it was
+ * disabled globally or on specification level.
+ *
+ * <p>The exact effect of this annotation depends on the further properties set on it as described
+ * in the documentation of the individual properties.
+ *
+ * <p>On any class that is not a specification and on any method that is not a feature, this annotation
+ * is simply ignored and has no effect.
+ *
+ * <p>If this annotation is applied on the same element as {@link DisableWeld @DisableWeld}, an exception
+ * is thrown as it is unclear which should have precedence.
+ *
+ * @author Björn Kautler
+ * @see DisableWeld
+ * @see EnableWeldExtension
+ * @see org.jboss.weld.spock.impl.EnableWeldManualInterceptor
+ * @see org.jboss.weld.spock.impl.EnableWeldAutoInterceptor
+ */
+@Retention(RUNTIME)
+@Target({ TYPE, METHOD })
+@Documented
+public @interface EnableWeld {
+    /**
+     * Whether to use the {@link org.jboss.weld.spock.impl.EnableWeldManualInterceptor} ({@code false}) with which you mostly
```

Comment:
It is not visible to code, but I enabled package protected classes for JavaDoc and so the link does work.
Actually I wanted to avoid having too much documentation at the `automagic` method so I split it to the respective classes and linked to them.
If you prefer, I can also copy all the text from both classes into that method JavaDoc and remove the links.